rust-app/bin-src/main_nanos.rs

```diff
@@ -63,23 +68,47 @@ extern "C" fn sample_main() {
     // let mut parsers = mk_parsers();
     let mut states = ParsersState::NoState;
 
+    let mut idle_menu = RootMenu::new([ WELCOME_TEXT, "Exit"]);
+    let mut busy_menu = RootMenu::new(["Working...", "Cancel"]);
+
+    let menu = |states: &ParsersState, idle: &mut RootMenu<2>, busy: &mut RootMenu<2>| match states {
+        ParsersState::NoState => idle.show(),
+        _ => busy.show(),
+    };
+
+    menu(&states, &mut idle_menu, &mut busy_menu);
+
     use core::mem::size_of_val;
     info!("State struct uses {} bytes\n", size_of_val(&states));
     // with_parser_state!(parsers);
 
     loop {
-        // Draw some 'welcome' screen
-        ui::SingleMessage::new("W e l c o m e").show();
-
         // Wait for either a specific button push to exit the app
         // or an APDU command
         match comm.next_event() {
-            io::Event::Button(ButtonEvent::RightButtonRelease) => nanos_sdk::exit_app(0),
-            io::Event::Command(ins) => match handle_apdu(&mut comm, ins, &mut states) {
-                Ok(()) => comm.reply_ok(),
-                Err(sw) => comm.reply(sw),
+            io::Event::Command(ins) => {
+                trace!("Command recived");
+                match handle_apdu(&mut comm, ins, &mut states) {
+                    Ok(()) => comm.reply_ok(),
+                    Err(sw) => comm.reply(sw),
+                };
+                menu(&states, & mut idle_menu, & mut busy_menu);
+                trace!("Command done");
+            },
+            io::Event::Button(btn) => {
+                trace!("Button recived");
+                if let ParsersState::NoState = states {
```

Comment:
I saw a lot more usages of `match`, is it normal/okay to used `if let` in scenarios like this instead?

Reply:
Looks good to me, provided that it's really clearer and that the compiler error when other variants are added isn't desirable.